runtime/src/snapshot_utils.rs

```diff
@@ -2988,6 +2988,27 @@ pub fn purge_old_bank_snapshots(
     );
 }
 
+/// At startup, purge old (i.e. unusable) bank snapshots
+///
+/// Only a single bank snapshot could be needed at startup (when using fast boot), so
+/// retain the highest bank snapshot "post", and purge the rest.
+pub fn purge_old_bank_snapshots_at_startup(bank_snapshots_dir: impl AsRef<Path>) {
+    let bank_snapshots_pre = get_bank_snapshots_pre(&bank_snapshots_dir);
+    purge_bank_snapshots(&bank_snapshots_pre);
+
+    let mut bank_snapshots_post = get_bank_snapshots_post(&bank_snapshots_dir);
+    bank_snapshots_post.sort_unstable();
+    let highest_bank_snapshot_post = bank_snapshots_post.pop();
+    purge_bank_snapshots(&bank_snapshots_post);
```

Comment:
Not sure it's explicitly better, but it's a bit more concise:
```suggestion
    purge_old_bank_snapshots(&bank_snapshots_dir, 0, Some(BankSnapshotType::Pre));
    purge_old_bank_snapshots(&bank_snapshots_dir, 1, Some(BankSnapshotType::Post));
    let highest_bank_snapshot_post = get_bank_snapshots_post(&bank_snapshots_dir).pop();
```

Probably if we were to go this route, I'd prefer having `purge_old_bank_snapshots` split and return the retained snapshots?

Reply:
Fine to leave this as you've got it as well, just giving a potential idea for reusing the existing fn that is very similar.

Reply:
> [1]: I like the split-and-return-retained idea. To reduce changes in this PR, is it OK to not do that here?

yeah, 100%. I'd definitely prefer that to be a separate PR 😄